cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,410 @@
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import datetime
+import itertools
+from typing import Dict, List, Tuple, \
+    TYPE_CHECKING
+
+import numpy as np
+
+from cirq import protocols
+from cirq.work.observable_settings import InitObsSetting, _max_weight_observable, _max_weight_state, \
+    _MeasurementSpec
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def _get_real_coef(observable: 'cirq.PauliString'):
+    """Assert that a PauliString has a real coefficient and return it."""
+    coef = observable.coefficient
+    if not np.isclose(coef.imag, 0):
```

Comment:
Should we expose the atol parameter here?

Reply:
done

Reply:
done